apps/ejabberd/src/mod_roster_riak.erl

```diff
@@ -0,0 +1,192 @@
+%%%----------------------------------------------------------------------
+%%% File    : mod_roster_riak.erl
+%%% Author  : Piotr Nosek <piotr.nosek@erlang-solutions.com>
+%%% Purpose : mod_roster Riak backend
+%%%
+%%% MongooseIM, Copyright (C) 2015      Erlang Solutions Ltd.
+%%%
+%%%----------------------------------------------------------------------
+-module(mod_roster_riak).
+
+-include("mod_roster.hrl").
+-include("jlib.hrl").
+
+-behaviour(mod_roster).
+
+%% API
+-export([init/2,
+         transaction/1,
+         read_roster_version/2,
+         write_roster_version/4,
+         get_roster/2,
+         get_roster_by_jid_t/3,
+         get_subscription_lists/3,
+         roster_subscribe_t/4,
+         get_roster_by_jid_with_groups_t/3,
+         remove_user/2,
+         update_roster_t/4,
+         del_roster_t/3,
+         read_subscription_and_groups/3,
+         raw_to_record/2]).
+
+-define(ROSTER_BUCKET(LServer), {<<"rosters">>, LServer}).
+-define(VER_BUCKET(LServer), {<<"roster_versions">>, LServer}).
+
+-spec init(ejabberd:server(), list()) -> ok.
+init(_Host, _Opts) ->
+    ok. % Common Riak pool is used
+
+-spec transaction(F :: fun()) -> {aborted, Reason :: any()} | {atomic, Result :: any()}.
+transaction(F) ->
+    put(riak_roster_t, []),
+    put(riak_version_t, []),
+    Return = case catch F() of
+                 {'EXIT', Reason} ->
+                     {aborted, Reason};
+                 Result ->
+                     %% Commiting...
+                     commit_t_roster(),
+                     commit_t_version(),
+                     put(riak_roster_t, []),
+                     {atomic, Result}
+             end,
+    lists:foreach(fun(Key) -> erase({riak_roster_t, Key}) end, erase(riak_roster_t)),
+    erase(riak_version_t),
+    Return.
+
+%% --------------------- Inside "transactions" --------------------------------
+
+get_roster_by_jid_t(LUser, LServer, LJID) ->
+    case riakc_map:find({jlib:jid_to_binary(LJID), register}, get_t_roster(LUser, LServer)) of
+        {ok, ItemReg} ->
+            (unpack_item(ItemReg))#roster{ jid = LJID, name = <<>>, groups = [], xs = [] };
```

Comment:
This line is cool, but on the other hand I think it could be more readable. I think breaking it into two lines like these
```erlang
Unpacked  = unpack_item(ItemReg),
Unpacked#roster{...}
```
will make it much more readable.